dissect/target/plugins/os/windows/sam.py

```diff
@@ -388,8 +388,8 @@ def sam(self) -> Iterator[SamRecord]:
                 u_lmpw = v_data[v.lmpw_ofs : v.lmpw_ofs + v.lmpw_len]
                 u_ntpw = v_data[v.ntpw_ofs : v.ntpw_ofs + v.ntpw_len]
 
-                lm_hash = decrypt_single_hash(f.rid, samkey, u_lmpw, almpassword).hex()
-                nt_hash = decrypt_single_hash(f.rid, samkey, u_ntpw, antpassword).hex()
+                lm_hash = decrypt_single_hash(f.rid, samkey, u_lmpw, almpassword).hex() if u_lmpw else ''
```

Comment:
Might be nicer to put something like this in the `decrypt_single_hash` function itself:

```python
if not enc_hash:
    return b""
```

That way you don't have to do this `if` twice and it's more fault-tolerant for future uses.

Reply:
Sure, fixed